### PR TITLE
#162163226 Users should be able Signup/Register to Authors Haven

### PR DESCRIPTION
**Pull request by Kasulejoseph**
#### What does this PR do?
Implement user signup functionality to Authors Haven product

#### Description of Task to be completed?
- Users are able to register to author's haven via the signup form.
- Descriptive error messages are displayed on unsuccessful signup.
- A message requesting a user to confirm his/her email is displayed to the user.

#### How should this be manually tested?
- On authors haven application, navigate to `/signup`
- Provide your `username`,` email`, and `password` details and then click `Sign Up` button to register

#### Any background context you want to provide?
You need to verify your email to be able to login

#### What are the relevant pivotal tracker stories?
[#162163226](https://www.pivotaltracker.com/story/show/162163226)

##### signup form
screen shot 2019-01-17 at 15 28 55

##### invalid details
screen shot 2019-01-17 at 15 29 51

### CODE REVIEW

**Review by KicaRonaldOkello — COMMENTED** [review]
@Kasulejoseph , could you please resolve merge conflicts in your actions, components and reducers folders? You could follow our team convention on that:

- checkout to develop branch

- `git pull origin develop`


- checkout to your branch, then git rebase develop

**Comment by Kasulejoseph** [comment]
> @Kasulejoseph , could you please resolve merge conflicts in your actions, components and reducers folders? You could follow our team convention on that:
> 
> * checkout to develop branch
> * `git pull origin develop`
> * checkout to your branch, then git rebase develop

Thanks for the feedback, let me work it now.

**Review by joshtrigger — APPROVED** [review]

**Review by Nta1e — APPROVED** [review]
LGTM👍🏽

**Review by kafuuma — APPROVED** [review]
@Kasulejoseph  I cloned and tested the future and everything works well, thanks for the great work. However, we can consider adding some functionality for loading when a user is waiting for the server to respond so that we improve the user experience. But this is good work

**Review by bibangamba — APPROVED** [review]

File: src/views/notFound/notFound.scss [hunk 1]
```diff
@@ -1,5 +1,5 @@
 
 .not-found-display{
-      font-size:72;
-      font-weight: 'bold'
+    font-size:72;
+    font-weight: 'bold'
```
[the remark this answers is not in the document]
Comment (JaymesKat) [reply src/views/notFound/notFound.scss]: @Kasulejoseph Fix this.

File: src/views/notFound/notFound.scss [hunk 2]
```diff
@@ -1,5 +1,5 @@
 
 .not-found-display{
-      font-size:72;
-      font-weight: 'bold'
+    font-size:72;
```
[the remark this answers is not in the document]
Comment (JaymesKat) [reply src/views/notFound/notFound.scss]: @Kasulejoseph Fix this.

[the remark this answers is not in the document]
Comment (Kasulejoseph) [reply src/views/notFound/notFound.scss]: worked on it.

**Review by JaymesKat — APPROVED** [review]
Good work

**Pull request merged**